daemon/prune.go

```diff
@@ -150,3 +154,72 @@ func (daemon *Daemon) ImagesPrune(config *types.ImagesPruneConfig) (*types.Image
 
 	return rep, nil
 }
+
+// localNetworksPrune removes unused local networks
+func (daemon *Daemon) localNetworksPrune(config *types.NetworksPruneConfig) (*types.NetworksPruneReport, error) {
+	rep := &types.NetworksPruneReport{}
+	var err error
+	// When the function returns true, the walk will stop.
+	l := func(nw libnetwork.Network) bool {
+		nwName := nw.Name()
+		predefined := runconfig.IsPreDefinedNetwork(nwName)
+		if !predefined && len(nw.Endpoints()) == 0 {
```

Comment:
I understand. I am trying to see if we can avoid the call to `deleteNetwork` which searches for the network by name when we already have the network handle `nw`. And avoid to check for predefined in two places.

But I agree we can take on this later as an enhancement and not block the addition of the new UI. Given this is no critical path.